src/XRP/issued-currency-client.ts

```diff
@@ -860,4 +911,34 @@ export default class IssuedCurrencyClient {
       sender,
     )
   }
+
+  /**
+   * Calculates the sendMaxValue by applying the transferFee to the amount being sent and ensuring the total amount
+   * fits within maximum decimal precision.
+   * @see https://xrpl.org/currency-formats.html#issued-currency-precision
+   *
+   * @param amount The amount of issued currency to pay to the destination.
+   * @param transferFee The transfer fee associated with the issuing account, expressed as a percentage.
+   *                    (i.e. a value of .5 indicates a 0.5% transfer fee).
+   */
+  public calculateSendMaxValue(amount: string, transferFee: number): string {
+    if (transferFee < 0) {
+      throw new XrpError(
+        XrpErrorType.InvalidInput,
+        'Transfer Fee must be positive.',
+      )
+    }
+    const numericAmount = new BigNumber(amount)
+    const transferRate = new BigNumber(1 + transferFee / 100)
+
+    // calculate the total sendMaxValue with any number of decimal places.
+    const rawSendMaxValue: BigNumber = numericAmount.multipliedBy(transferRate)
+
+    if (rawSendMaxValue.toFixed().length <= 15) {
+      return rawSendMaxValue.toFixed()
+    }
+
+    // If we have too many digits of precision, express with scientific notation but fit within decimal precision.
+    return rawSendMaxValue.toExponential(14, 0).replace(/e\+/, 'e')
```

Comment:
might be good to make consts for these magic numbers

Reply:
yeah can do I'll call this thing maxDecimalPrecision or something